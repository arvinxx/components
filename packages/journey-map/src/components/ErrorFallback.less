@import '../token';

.avx-journey-error {
  &-detail {
    margin-top: 8px;
    .secondText;
  }
}
