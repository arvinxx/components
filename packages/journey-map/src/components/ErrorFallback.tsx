import React from 'react';
import { Alert, Typography } from 'antd';

import './ErrorFallback.less';

const { Title, Text } = Typography;
const ErrorFallback = ({ error }) => (
  <Alert
    type={'error'}
    message={
      <div>
        <Title level={5} className="avx-journey-error-title">
          渲染出错啦 😮
        </Title>
        <Text className="avx-journey-error-desc">
          请检查输入语法是否存在问题，错误详情如下:
        </Text>
        <pre>
          <Text ellipsis copyable className="avx-journey-error-detail">
            {error.message}
          </Text>
        </pre>
        <Text className="avx-journey-error-desc">
          若检查后无语法错误，请
          <a
            href="https://github.com/arvinxx/components/issues/new"
            style={{ margin: '0 4px' }}
          >
            点击这里
          </a>
          提交 bug
        </Text>
      </div>
    }
  />
);
export default ErrorFallback;
