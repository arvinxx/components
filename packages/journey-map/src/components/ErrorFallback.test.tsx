import React from 'react';
import { render } from '@testing-library/react';

import ErrorFallback from './ErrorFallback';

describe('ErrorFallback', () => {
  it('默认', () => {
    const { container } = render(<ErrorFallback error={'123'} />);
    expect(container).toMatchSnapshot();
  });
});
