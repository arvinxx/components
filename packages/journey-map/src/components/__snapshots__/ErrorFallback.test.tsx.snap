// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorFallback 默认 1`] = `
<div>
  <div
    class="ant-alert ant-alert-error ant-alert-no-icon"
    data-show="true"
    role="alert"
  >
    <div
      class="ant-alert-content"
    >
      <div
        class="ant-alert-message"
      >
        <div>
          <h5
            class="ant-typography avx-journey-error-title"
          >
            渲染出错啦 😮
          </h5>
          <span
            class="ant-typography avx-journey-error-desc"
          >
            请检查输入语法是否存在问题，错误详情如下:
          </span>
          <pre>
            <span
              class="ant-typography ant-typography-ellipsis ant-typography-single-line avx-journey-error-detail"
            >
              <div
                aria-label="Copy"
                class="ant-typography-copy"
                role="button"
                style="border: 0px; background: transparent; padding: 0px; line-height: inherit; display: inline-block;"
                tabindex="0"
              >
                <span
                  aria-label="copy"
                  class="anticon anticon-copy"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="copy"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"
                    />
                  </svg>
                </span>
              </div>
              <span
                aria-hidden="true"
                style="position: fixed; display: block; left: 0px; top: 0px; z-index: -9999; visibility: hidden; pointer-events: none; font-size: 0px; word-break: keep-all; white-space: nowrap;"
              >
                lg
              </span>
              <span
                aria-hidden="true"
                style="position: fixed; display: block; left: 0px; top: 0px; z-index: -9999; visibility: hidden; pointer-events: none; font-size: 0px; width: 0px; white-space: normal; margin: 0px; padding: 0px;"
              >
                <span
                  aria-hidden="true"
                >
                  ...
                </span>
                <div
                  aria-label="Copy"
                  class="ant-typography-copy"
                  role="button"
                  style="border: 0px; background: transparent; padding: 0px; line-height: inherit; display: inline-block;"
                  tabindex="0"
                >
                  <span
                    aria-label="copy"
                    class="anticon anticon-copy"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="copy"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"
                      />
                    </svg>
                  </span>
                </div>
              </span>
            </span>
          </pre>
          <span
            class="ant-typography avx-journey-error-desc"
          >
            若检查后无语法错误，请
            <a
              href="https://github.com/arvinxx/components/issues/new"
              style="margin: 0px 4px;"
            >
              点击这里
            </a>
            提交 bug
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
